### PR TITLE
Lasb 956 assessment summary rules

### PR DESCRIPTION
**Pull request by AafAnsari**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-956)

- added new service class, dtos, enums and unit test for adding assessment summary to create means response
- added url config in application.yml and updated corresponding class
- updated response schema with assessmentSummary
- some minor refactoring

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentSummaryService.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+public class AssessmentSummaryService {
+
+    private final MaatCourtDataService maatCourtDataService;
+
+    public ApiCreateMeansAssessmentResponse addAssessmentSummaryToMeansResponse(final ApiCreateMeansAssessmentResponse assessmentResponse,
+                                                                                final String laaTransactionId) {
+        log.info("Generating assessment summary for means assessment response");
+        try {
+            var assesmentsSummary = new ArrayList<ApiAssessmentSummary>();
+            assesmentsSummary.add(getFinancialAssessmentSummary(assessmentResponse));
```
Comment (mtac50) [inline crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentSummaryService.java:32]: `assesmentsSummary` should be `assessmentsSummary`

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentSummaryService.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    private ApiAssessmentSummary getFinancialAssessmentSummary(final ApiCreateMeansAssessmentResponse assessmentResponse) {
+        log.info("Generating assessment summary for financial assessment");
+        try {
+            ApiAssessmentSummary finAssessmentSummary = new ApiAssessmentSummary();
+            finAssessmentSummary.withId(assessmentResponse.getAssessmentId())
+                    .withReviewType(ofNullable(assessmentResponse.getReviewType()).map(ReviewType::getCode).orElse(null));
+
+            if (assessmentResponse.getAssessmentType() == AssessmentType.INIT) {
+                finAssessmentSummary.setType(WorkType.Initial_Assessment);
+                finAssessmentSummary.setStatus(ofNullable(assessmentResponse.getFassInitStatus()).map(CurrentStatus::getDescription)
```
Comment (mtac50) [inline crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentSummaryService.java:52]: Can the `setStatus` call be brought outside of the `if/else`?

File: crime-means-assessment/src/test/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentSummaryServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        verify(maatCourtDataService).getIOJAppealFromRepId(1234, null);
+    }
+
+    @Test
+    public void givenMeansAssessmentResponseWithFullAssDate_WhenGetAssessmentSummaryIsInvoked_ThenAssessmentSummaryShouldBeReturnedWithFullFinAssSummary() {
+        meansAssessmentResponse.setAssessmentType(AssessmentType.FULL);
+        meansAssessmentResponse.setFullAssessmentDate(LocalDateTime.of(2022, 1, 20, 10, 0));
+        assessmentSummaryService.addAssessmentSummaryToMeansResponse(meansAssessmentResponse, null);
+
+        Assertions.assertThat(meansAssessmentResponse.getAssessmentSummary()).extracting("type")
```
Comment (mtac50) [inline crime-means-assessment/src/test/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentSummaryServiceTest.java:76]: Is it worth adding in an additional assertion to check the correct assessment data has been set in the `getFinancialAssessmentSummary` logic? (The same would go for the previous two tests as well)

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentSummaryService.java (hunk 1 above)
Comment (AafAnsari) [reply crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentSummaryService.java:32]: yes typo. done

File: crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentSummaryService.java (hunk 2 above)
Comment (AafAnsari) [reply crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentSummaryService.java:52]: No, condition is in place to set InitStatus and FullStatus

File: crime-means-assessment/src/test/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentSummaryServiceTest.java (hunk 3 above)
Comment (AafAnsari) [reply crime-means-assessment/src/test/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentSummaryServiceTest.java:76]: yes definitely, I'll add asserts on date field

**Review by mtac50 — APPROVED** [review]

**Pull request merged**